tests/countries/test_guinea.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_africa_day(self):
+        self.assertHolidayName(
+            "Anniversaire de l'OUA", (f"{year}-05-25" for year in range(1958, 2050))
+        )
+
+    def test_eid_al_fitr_day(self):
+        name = "Korité"
+        self.assertHolidayName(
+            name,
+            "2015-07-18",
+            "2016-07-07",
+            "2017-06-26",
+            "2018-06-15",
+            "2019-06-04",
+            "2020-05-24",
+            "2021-05-13",
+            "2022-05-02",
+            "2023-04-21",
+            "2024-04-10",
+            "2025-03-31",
+        )
+
+    def test_easter_monday(self):
+        name = "Le lundi de Pâques"
+        self.assertHolidayName(
+            name,
+            "2015-04-06",
+            "2016-03-28",
+            "2017-04-17",
+            "2018-04-02",
+            "2019-04-22",
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+        )
+
+    def test_labor_day(self):
+        self.assertHolidayName("Fête du Travail", (f"{year}-05-01" for year in range(1958, 2050)))
+
+    def test_assumption_of_mary(self):
+        name = "Assomption de Marie"
+        self.assertHolidayName(name, (f"{year}-08-15" for year in range(1958, 2050)))
+
+    def test_independence_day(self):
+        self.assertHolidayName(
+            "Fête de l'indépendance de la Guinée", (f"{year}-10-02" for year in range(1958, 2050))
+        )
+
+    def test_christmas_day(self):
+        self.assertHolidayName("Noël", (f"{year}-12-25" for year in range(1958, 2050)))
+
+    def test_mawlid_day(self):
+        name = "Maouloud"
+        self.assertHolidayName(
+            name,
+            "2015-12-24",
+            "2016-12-12",
+            "2017-12-01",
+            "2018-11-20",
+            "2019-11-10",
+            "2020-10-29",
+            "2021-10-18",
+            "2022-10-08",
+            "2023-09-27",
+            "2024-09-15",
+            "2025-09-05",
+        )
+
+    def test_eid_al_adha_day(self):
+        name = "Tabaski"
+        self.assertHolidayName(
+            name,
+            "2015-09-24",
+            "2016-09-13",
+            "2017-09-02",
+            "2018-08-22",
+            "2019-08-11",
+            "2020-07-31",
+            "2021-07-20",
+            "2022-07-10",
+            "2023-06-28",
+            "2024-06-16",
+            "2025-06-07",
+        )
+
+    def test_laylat_al_qadr_day(self):
+        name = "Lailatoul Qadr"
+        self.assertHolidayName(
+            name,
+            "2015-07-14",
+            "2016-07-03",
+            "2017-06-22",
+            "2018-06-12",
+            "2019-06-01",
+            "2020-05-20",
+            "2021-05-09",
+            "2022-04-29",
+            "2023-04-18",
+            "2024-04-06",
+            "2025-03-27",
+        )
+
+    def test_2021(self):
+        self.assertHolidays(
```

Comment:
Maybe consider adding link to source you used in test case as well for `test_[year]` test cases:
```suggestion
    def test_2021(self):
        # insert-source-link-here
        self.assertHolidays(
```